Automattic/wp-calypso
# Developer UX: Scroll to top on mount Performance Settings and Hosting Configuration

- Closes https://github.com/Automattic/dotcom-forge/issues/1635

## Proposed Changes

* Create a new component `ScrollToTopOnMount` and a new prop on `Main`
* Use this new prop on `Performance settings` and `Hosting configuration` to fix the old scroll position when we navigate from SMP to one of those pages.

## Testing Instructions



* Apply this branch and try it on localhost or wpcalypso.
* Go to SMP `/sites`
* Scroll down the page
* Click on the actions menu on a business site
* Click on `Performance settings` or `Hosting configuration`
* Observe you've navigated to the correct page and the scroll is on the top. Before this PR the scroll would remain in the middle of the page.

| **Before** | **After** |
|---|---|
| ![scroll-before](https://user-images.githubusercontent.com/779993/221303165-dd81a564-e328-455b-aab8-c25faaf685f8.gif) | ![scroll-after](https://user-images.githubusercontent.com/779993/221303170-d3c5b0ac-2fc4-47cc-ba02-d0a7fe8080da.gif) |

## Pre-merge Checklist
